packages/cojson/src/coValues/group.ts

```diff
@@ -258,29 +258,38 @@ export class RawGroup<
     role: Role,
   ) {
     if (account === EVERYONE) {
-      if (!(role === "reader" || role === "writer")) {
+      if (!(role === "reader" || role === "writer" || role === "writeOnly")) {
         throw new Error(
-          "Can't make everyone something other than reader or writer",
+          "Can't make everyone something other than reader, writer or writeOnly",
         );
       }
-
       const currentReadKey = this.core.getCurrentReadKey();
 
       if (!currentReadKey.secret) {
         throw new Error("Can't add member without read key secret");
       }
 
+      const previousRole = this.get(account);
+
       this.set(account, role, "trusting");
 
       if (this.get(account) !== role) {
         throw new Error("Failed to set role");
       }
 
-      this.set(
-        `${currentReadKey.id}_for_${EVERYONE}`,
-        currentReadKey.secret,
-        "trusting",
-      );
+      if (role === "writeOnly") {
+        if (previousRole === "reader" || previousRole === "writer") {
+          this.rotateReadKey();
+        }
```

Comment:
Changing from reader/writer to writeOnly means that we remove the read rights, hence we need to rotate keys.